Added e2e tests for page.added webhook

JohnGrisham opened the pull request:
Added e2e tests for page.added webhook

Ref - https://github.com/TryGhost/Ghost/issues/15537

naz commented on ghost/core/test/e2e-webhooks/pages.test.js:
```diff
@@ -0,0 +1,122 @@
+    anyNumber
+} = matchers;
+
+const anyGhostAgent = stringMatching(
```
The `anyGhostAgent` matcher is not a part of "matcher", so there's no need to re-declare it here 🙃  If you rebase your changes over the latest `main` the matcher should be available 👍

JohnGrisham replied:
Thanks!

naz commented on ghost/core/test/e2e-webhooks/pages.test.js:
```diff
@@ -0,0 +1,122 @@
+
+    // NOTE: this is such a bad hack! for the reasons I did not investigate the "add" event does not include
+    //       the roles but the "published" does! massive inconsistency and needs to be fixed one day
+    if (roles) {
```
The 'roles' here does not seem to ever be "true", so probably no need for this whole block here

naz reviewed (changes requested):
This is great! Thanks @JohnGrisham . I've requested couple very minor changes, otherwise it's ready to land in main 👍

naz reviewed (approved):
The changes are good to go into main. Thanks @JohnGrisham 🙌  
A little note, could you please make sure the changes come squashed (as a single commit) and the commit message follows Ghost's [contributor guidelines](https://github.com/TryGhost/Ghost/blob/main/.github/CONTRIBUTING.md#commit-messages) 🙏

Commit:
Added end to end test case for the "page added" webhook event.

refs #15537
snapshot test created to add confidence to webhook stability and increase overall test coverage.

JohnGrisham commented:
Changes were made and the commits consolidated into one. Thanks for your help!

ErisDS commented:
Hi @JohnGrisham thanks so much for this PR. It has now been merged 🎉 

I'm not sure if you found this through hacktoberfest, but I've added the accepted label to this PR to make sure it counts. We have plenty of open issues and ongoing projects for hacktoberfest and beyond. We would love to see you around again 👋
